### PR TITLE
sci-libs/vtk: version bump to 9.0.1 + dependencies

### PR DESCRIPTION
**Pull request by epsilon-0**
Still a WIP...

---

Depends on https://github.com/gentoo/gentoo/pull/18767
Depends on https://bugs.gentoo.org/761439
Depends on https://bugs.gentoo.org/761460

---

Even better, all `VTK_USE_SYSTEM_<PKG>` have been removed and are called (I think) `VTK_MODULE_USE_EXTERNAL_<PKG>`

### CODE REVIEW

**Comment by a17r** [comment]
See also https://github.com/gentoo/gentoo/pull/15966

File: sci-libs/vtk/vtk-9.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+		-DBUILD_EXAMPLES=$(usex examples)
+		-DVTK_MODULE_USE_EXTERNAL_utf8cpp=ON
+		-DVTK_USE_SYSTEM_AUTOBAHN=ON
+		-DVTK_USE_SYSTEM_EXPAT=ON
+		-DVTK_USE_SYSTEM_FREETYPE=ON
+		-DVTK_USE_SYSTEM_FreeType=ON
+		-DVTK_USE_SYSTEM_GL2PS=ON
+		-DVTK_USE_SYSTEM_HDF5=ON
+		-DVTK_USE_SYSTEM_JPEG=ON
+		-DVTK_USE_SYSTEM_LIBPROJ=OFF
```
Comment (Alessandro-Barbieri) [inline sci-libs/vtk/vtk-9.0.1.ebuild:190]: Why depend on proj if isn't used? Is it possible to enable?

File: sci-libs/vtk/vtk-9.0.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+		-DVTK_USE_COCOA=$(usex aqua)
+	)
+
+	if use java; then
+		local javacargs=$(java-pkg_javac-args)
+		mycmakeargs+=( -DJAVAC_OPTIONS=${javacargs// /;} )
+	fi
+
+	if use mpi; then
+		mycmakeargs+=( -DVTK_USE_SYSTEM_MPI4PY=ON )
```
Comment (Alessandro-Barbieri) [inline sci-libs/vtk/vtk-9.0.1.ebuild:252]: I think this should be nested into the if use python below
See also line 92

Comment (epsilon-0) [reply sci-libs/vtk/vtk-9.0.1.ebuild:252]: There are so many changes to make, reviewing and commenting right now is not that useful.
Basically, most options and workings are going to be replaced.

Comment (thesamesam) [reply sci-libs/vtk/vtk-9.0.1.ebuild:252]: Yeah, I take non-draft to mean "ready for comments". Let her work for now. I use GitHub PRs like this for CI and to keep the work safe.

**Pull request closed without merging**